feat: sign linux system packages with GPG identities (#2973)
beeware/briefcase: changes/2396.feature.md

> Linux system packages (`.deb`, `.rpm` and `.pkg.tar.zst`) can now be signed with a GPG signing identity, using the `--identity` option to `briefcase package`. If no identity is specified, Briefcase will prompt you to select a signing identity from the secret keys available on your system, or to opt out of signing. Signing is not supported when building with Docker; packages built with the `--target` option must be produced without a signature.

Reviewer: This is for the release notes - it doesn't need to be a full usage guide.
```suggestion
Linux system packages (`.deb`, `.rpm` and `.pkg.tar.zst`) can now be signed with a GPG signing identity, using the `--identity` option to `briefcase package`. Signing is not currently supported when building with Docker.
```
